micropip/_commands/freeze.py

```diff
@@ -42,23 +42,23 @@ def freeze_data() -> dict[str, Any]:
 def load_pip_packages() -> Iterator[tuple[str, dict[str, Any]]]:
     return map(
         package_item,
-        filter(None, map(load_pip_package, importlib.metadata.distributions())),
+        filter(is_valid, map(load_pip_package, importlib.metadata.distributions())),
     )
 
 
 def package_item(entry: dict[str, Any]) -> tuple[str, dict[str, Any]]:
     return canonicalize_name(entry["name"]), entry
 
 
-def load_pip_package(dist: importlib.metadata.Distribution) -> dict[str, Any] | None:
+def is_valid(entry: dict[str, Any]) -> bool:
+    return entry["url"] is not None
+
+
+def load_pip_package(dist: importlib.metadata.Distribution) -> dict[str, Any]:
     name = dist.name
     version = dist.version
     url = dist.read_text("PYODIDE_URL")
-    if url is None:
-        return
-
     sha256 = dist.read_text("PYODIDE_SHA256")
-    assert sha256
```

Comment:
I think mypy will be mad about this since the type of sha256 is `str | None` and this assert is here to make mypy change the type to str.

Reply:
It seems like mypy passed?

Reply:
> I think mypy will be mad about this since the type of sha256 is `str | None` and this assert is here to make mypy change the type to str.

I don't understand the problem. `read_text` returns `str | None` and `sha256` is `str | None`, so there's no need to cast it to `str`.